packages/near-membrane-base/src/membrane.ts

```diff
@@ -0,0 +1,1549 @@
+            this.makeProxyUnambiguous(shadowTarget);
+            return this.defineProperty?.(shadowTarget, key, partialDesc);
+        }
+
+        // static default traps (optimization to avoid computations of the proper
+        // trap in constructor)
+        private static defaultOwnKeysTrap = BoundaryProxyHandler.dynamicOwnKeysTrap;
+
+        private static defaultIsExtensibleTrap = BoundaryProxyHandler.dynamicIsExtensibleTrap;
+
+        private static defaultGetOwnPropertyDescriptorTrap =
+            BoundaryProxyHandler.dynamicGetOwnPropertyDescriptorTrap;
+
+        private static defaultGetPrototypeOfTrap = BoundaryProxyHandler.dynamicGetPrototypeOfTrap;
+
+        private static defaultGetTrap = BoundaryProxyHandler.dynamicGetTrap;
+
+        private static defaultHasTrap = BoundaryProxyHandler.dynamicHasTrap;
+
+        // pending traps are only really needed if this membrane
+        // traps mutations to avoid mutations operations on the
+        // side of the membrane.
+        // TODO: find a way to optimize the declaration rather than instantiation
+        private static defaultSetPrototypeOfTrap = trapMutations
+            ? BoundaryProxyHandler.pendingSetPrototypeOfTrap
+            : BoundaryProxyHandler.dynamicSetPrototypeOfTrap;
+
+        private static defaultSetTrap = trapMutations
+            ? BoundaryProxyHandler.pendingSetTrap
+            : BoundaryProxyHandler.dynamicSetTrap;
+
+        private static defaultDeletePropertyTrap = trapMutations
+            ? BoundaryProxyHandler.pendingDeletePropertyTrap
+            : BoundaryProxyHandler.dynamicDeletePropertyTrap;
+
+        private static defaultPreventExtensionsTrap = trapMutations
+            ? BoundaryProxyHandler.pendingPreventExtensionsTrap
+            : BoundaryProxyHandler.dynamicPreventExtensionsTrap;
+
+        private static defaultDefinePropertyTrap = trapMutations
+            ? BoundaryProxyHandler.pendingDefinePropertyTrap
+            : BoundaryProxyHandler.dynamicDefinePropertyTrap;
+    }
+    setPrototypeOf(BoundaryProxyHandler.prototype, null);
+
+    // future optimization: hoping that proxies with frozen handlers can be faster
+    freeze(BoundaryProxyHandler.prototype);
+
+    function linkGlobalThis(foreignGlobalThisPointers: Pointer) {
+        WeakMapSet.call(proxyTargetToPointerMap, globalThis, foreignGlobalThisPointers);
+    }
+
+    // These are foundational things that should never be wrapped but are equivalent
+    // TODO: revisit this list.
+    const ReflectiveIntrinsicObjectNames = [
+        'AggregateError',
+        'Array',
+        'Error',
+        'EvalError',
+        'Function',
+        'Object',
+        'Proxy',
+        'RangeError',
+        'ReferenceError',
+        'SyntaxError',
+        'TypeError',
+        'URIError',
+    ];
+
+    const reflectivePointers: Pointer[] = [];
+    const reflectiveValues: any[] = [];
+    ReflectiveIntrinsicObjectNames.forEach((globalName) => {
```

Comment:
`.forEach`

Reply:
this one doesn't matter much because it is initialization code... that runs first... caching the method will not really help here